sql/field.h

```diff
@@ -4315,6 +4317,13 @@ class Create_field {
 
   Nullable<gis::srid_t> m_srid;
 
+  /*
+    Store dict_id after verifying zip_dict_name exists. The stored id
+    is filled in dd::Column::options and later used to fill TABLE_SHARE*
+    zip_dict_name and zip_dict_data
+  */
+  uint64 zip_dict_id;
```

Comment:
When the id is retrieved from DD, it is dd::Object_id which is 
typedef unsigned long long Object_id;

 There might be compilation warnings because of this change.